### PR TITLE
Carla camera streaming integration

### PR DESCRIPTION
**Pull request by DKlimenko**
Carla camera streaming integration

### CODE REVIEW

**Comment by arminn** [comment]
@svlad-90 @dterletskiy plz review

**Comment by svlad-90** [comment]
Hi @DKlimenko, please:

- add "Signed-off-by: <dmytro_klimenko@epam.com>" to your commit messages
- add a proper commit message to each commit which describes what was changed and the motivation for the change
- squash redundant commits

File: build/device.mk [hunk 1]
```diff
@@ -380,6 +380,6 @@ $(call inherit-product, device/xen/xenvm/build/kernel_modules.mk)
 $(call inherit-product, vendor/epam/aosp-hmi/epam-automotive-launcher.mk)
 $(call inherit-product, vendor/epam/ces-navigation/epam-navigation.mk)
 #$(call inherit-product, vendor/epam/EpamCluster/epam-cluster-product.mk)
-$(call inherit-product, vendor/epam/EpamMusic/epam-audio.mk)
+#$(call inherit-product, vendor/epam/EpamMusic/epam-audio.mk)
```
Comment (svlad-90) [inline build/device.mk]: Hi @DKlimenko, you've uncommented these lines here: https://github.com/xen-troops/android_device_xenvm/pull/92/commits/9afe714e2d776e7b60eb962f5a65337eb31b122c

Why do you need this change as a separate commit? I think it's not valid. Please, remove your "experiments", and leave only a set of final valuable changes.

File: build/device.mk [hunk 2]
```diff
@@ -381,5 +381,5 @@ $(call inherit-product, vendor/epam/aosp-hmi/epam-automotive-launcher.mk)
 $(call inherit-product, vendor/epam/ces-navigation/epam-navigation.mk)
 #$(call inherit-product, vendor/epam/EpamCluster/epam-cluster-product.mk)
 $(call inherit-product, vendor/epam/EpamMusic/epam-audio.mk)
-$(call inherit-product, vendor/epam/EpamHNAV/epam-hnav.mk)
+#$(call inherit-product, vendor/epam/EpamHNAV/epam-hnav.mk)
```
Comment (svlad-90) [inline build/device.mk]: Hi @DKlimenko, you've uncommented this line here: https://github.com/xen-troops/android_device_xenvm/pull/92/commits/9afe714e2d776e7b60eb962f5a65337eb31b122c

Why do you need this change as a separate commit? I think it's not valid. Please, remove your "experiments", and leave only a set of final valuable changes.

File: build/device.mk [hunk 3]
```diff
@@ -379,7 +379,7 @@ $(call inherit-product, device/xen/xenvm/build/security.mk)
 $(call inherit-product, device/xen/xenvm/build/kernel_modules.mk)
 $(call inherit-product, vendor/epam/aosp-hmi/epam-automotive-launcher.mk)
 $(call inherit-product, vendor/epam/ces-navigation/epam-navigation.mk)
-$(call inherit-product, vendor/epam/EpamCluster/epam-cluster-product.mk)
+#$(call inherit-product, vendor/epam/EpamCluster/epam-cluster-product.mk)
```
Comment (svlad-90) [inline build/device.mk]: Hi @DKlimenko, you've uncommented this line here: https://github.com/xen-troops/android_device_xenvm/pull/92/commits/9afe714e2d776e7b60eb962f5a65337eb31b122c

Why do you need this change as a separate commit? I think it's not valid. Please, remove your "experiments", and leave only a set of final valuable changes.

File: build/device.mk [hunk 4]
```diff
@@ -278,9 +278,6 @@ ENABLE_CAMERA_SERVICE=true
 ENABLE_EVS_SAMPLE=true
 $(call inherit-product, vendor/xen/evs/evs.mk)
 
-PRODUCT_PRODUCT_PROPERTIES += \
```
Comment (svlad-90) [inline build/device.mk]: Hi @DKlimenko, you've added this directive here in the same PR - https://github.com/xen-troops/android_device_xenvm/pull/92/commits/ec9935a0be8106cf948fb88f1c7ba01810167a03#diff-a77da357f7efbbcf77d2baa145a6bb6da7bd54e2a956b94727149dce6849d107R281

But now you remove it within the same PR. It is not valid. Please, squash not-needed lines to not pollute the diff with useless changes.

**Review by svlad-90 — CHANGES_REQUESTED** [review]

File: build/device.mk [hunk 5]
```diff
@@ -272,6 +272,16 @@ PRODUCT_COPY_FILES += \
     vendor/xen/sensors/cfg/sensors-config.json:$(TARGET_COPY_OUT_VENDOR)/etc/vehicle/sensors-config.json
 
 # Exterior View System (EVS)
+USE_CARLA_EVS=true
```
Comment (svlad-90) [inline build/device.mk]: Hi @DKlimenko, maybe you should use the "USE_CARLA_EVS?=true" to allow overwriting this value?

Otherwise, what is the reason for having a condition at line 276 for the variable, which is always equal to true?

**Comment by DKlimenko** [comment]
> Hi @DKlimenko, please:
> 
> * add "Signed-off-by: [dmytro_klimenko@epam.com](mailto:dmytro_klimenko@epam.com)" to your commit messages
> * add a proper commit message to each commit which describes what was changed and the motivation for the change
> * squash redundant commits

Hi. Squashed commits. 
Was thinking PR supports squashing.
Signed-off added

File: build/device.mk (hunk 5 above)
Comment (DKlimenko) [reply build/device.mk]: It is our flag. Requested by Andrey

Comment (svlad-90) [reply build/device.mk]: Hi @DKlimenko, but shouldn't it be used with "?=". Could you, please, explain why it is needed? What is the logical reason for it?

Comment (DKlimenko) [reply build/device.mk]: Hi @svlad-90, it is done for future compile flag integration for EVS functionality

Comment (arminn) [reply build/device.mk]: This flag should be set externally by moulin or shell. Since it is a generic code and we at least now don't want to enable it all the time.

File: build/device.mk [hunk 6]
```diff
@@ -272,6 +272,16 @@ PRODUCT_COPY_FILES += \
...
+ifeq ($(USE_CARLA_EVS), true)
+ENABLE_CAMERA_SERVICE=true
+ENABLE_EVS_SAMPLE=true
+$(call inherit-product, vendor/xen/evs/evs.mk)
+
+PRODUCT_PROPERTY_OVERRIDES += \
+	persist.vendor.evs.uri="ws://192.168.122.1"
```
Comment (arminn) [inline build/device.mk]: I suppose here we need to use a host name(like we did previously with wwwivi) instead of IP. Since our setup configuration may change and will resolve host using static record in /etc/hosts

**Review by arminn — CHANGES_REQUESTED** [review]
@DKlimenko plz also add meaningful description to patch : what was done, why and etc.

**Comment by DKlimenko** [comment]
> @DKlimenko plz also add meaningful description to patch : what was done, why and etc.

Done.

File: build/device.mk (hunk 6 above)
Comment (DKlimenko) [reply build/device.mk]: Not sure if we need to add carla host to /etc/hosts

Comment (DKlimenko) [reply build/device.mk]: as was agreed with @arminn flag was removed

File: build/device.mk (hunk 1 above)
Comment (DKlimenko) [reply build/device.mk]: There was HMI error, I just commented out it. In final commit it is unchanged

File: build/device.mk (hunk 2 above)
Comment (DKlimenko) [reply build/device.mk]: Hi. Squashed commits. Not actual any more

File: build/device.mk (hunk 3 above)
Comment (DKlimenko) [reply build/device.mk]: Hi. Squashed commits. Not actual any more

File: build/device.mk (hunk 4 above)
Comment (DKlimenko) [reply build/device.mk]: Hi. Squashed commits. Not actual any more

File: build/device.mk (hunk 5 above)
Comment (DKlimenko) [reply build/device.mk]: As was agreed with @arminn , flag was removed.
